config/m4/compiler.m4

```diff
@@ -449,6 +449,27 @@ AS_IF([test "x$enable_gcov" = xyes],
                                        [AC_LANG_SOURCE([[int main(int argc, char** argv){return 0;}]])])],
       [:])
 
+#
+# Enable stack usage check
+#
+# To suppress -Wframe-larger-than=8192 for specific functions, use:
+# #pragma GCC diagnostic push
+# #pragma GCC diagnostic ignored "-Wframe-larger-than="
+# void specific_function() {
+#     /* Function code */
+# }
+# #pragma GCC diagnostic pop
+#
+AC_ARG_ENABLE([stack-usage-check],
+        AS_HELP_STRING([--enable-stack-usage-check], [Enable stack usage check with -Wframe-larger-than=8192]),
+        [],
+        [enable_stack_usage_check=yes])
+
+AS_IF([test "x$enable_stack_usage_check" = xyes && test "x$enable_asan" != xyes],
```

Comment:
use "-a" instead of "&&", see other usages of AS_IF in the codebase

Reply:
Changed `&&` to nested `AS_IF`